packages/edit-site/src/hooks/use-theme-style-variations/use-theme-style-variations-by-property.js

```diff
@@ -1,13 +1,76 @@
 /**
  * WordPress dependencies
  */
-import { useMemo } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import { useContext, useMemo } from '@wordpress/element';
+import { privateApis as blockEditorPrivateApis } from '@wordpress/block-editor';
 
 /**
  * Internal dependencies
  */
 import { mergeBaseAndUserConfigs } from '../../components/global-styles/global-styles-provider';
 import cloneDeep from '../../utils/clone-deep';
+import { unlock } from '../../lock-unlock';
+
+const { GlobalStylesContext } = unlock( blockEditorPrivateApis );
+
+/**
+ * Removes all instances of a property from an object.
+ *
+ * @param {Object} object   The object to remove the property from.
+ * @param {string} property The property to remove.
+ * @return {Object} The modified object.
+ */
+export function removePropertyFromObject( object, property ) {
```

Comment:
I deleted this function by accident. So just reverting it and adding tests.

It's required to ensure that variations remain "pure". 

Without this, user-defined properties will persist in any merge. For example, if a user has defined a background color and the selected variation has no background color, the user-defined color will persist. 

This is best demonstrated in the 2023 theme. Notice how the gradient backgrounds are merged with other variations:

https://github.com/WordPress/gutenberg/assets/6458278/72fb8ccc-edc9-493c-aedc-b28d91145c0f

Reply:
Thanks I had noticed the bug but I didn't know the cause :)